gems/pending/spec/appliance_console/database_maintenance_spec.rb

```diff
@@ -0,0 +1,99 @@
+require "appliance_console/prompts"
+require "appliance_console/database_maintenance"
+require "fileutils"
+require "tempfile"
+require "tmpdir"
+
+describe ApplianceConsole::DatabaseMaintenance do
+  DIRNAME = File.dirname(__FILE__).freeze
+
+  before do
+    allow(subject).to receive(:say)
+  end
+
+  describe "#ask_questions" do
+    before do
+      allow(subject).to receive(:say)
+      allow(subject).to receive(:clear_screen)
+      allow(subject).to receive(:agree)
+    end
+
+    context "when an hourly cron job does not exist" do
+      before do
+        @test_hourly_cron = "#{Dir.tmpdir}/miq-pg-maintenance-hourly.cron"
+        stub_const("ApplianceConsole::DatabaseMaintenance::HOURLY_CRON", @test_hourly_cron)
+      end
+
+      it "returns true when configure is confirmed" do
+        expect(subject).to receive(:agree).with(/configure database maintenance/i).and_return(true)
+        expect(subject.ask_questions).to be true
+      end
+
+      it "returns false when configure is not confirmed" do
+        expect(subject).to receive(:agree).with(/configure database maintenance/i).and_return(false)
+        expect(subject.ask_questions).to be false
+      end
+    end
+
+    context "when an hourly cron job does exist" do
+      before do
+        @test_hourly_cron = Tempfile.new(subject.class.name.split("::").last.downcase)
+        stub_const("ApplianceConsole::DatabaseMaintenance::HOURLY_CRON", @test_hourly_cron.path)
+      end
+
+      after do
+        FileUtils.rm_f(@test_hourly_cron.path)
+      end
+
+      it "returns true when un-configure is confirmed" do
+        expect(subject).to receive(:agree).with(/un-configure/i).and_return(true)
+        expect(subject.ask_questions).to be true
+      end
+
+      it "returns false when un-configure is not confirmed" do
+        expect(subject).to receive(:agree).with(/un-configure/i).and_return(false)
+        expect(subject.ask_questions).to be false
+      end
+    end
+  end
+
+  describe "#activate" do
+    context "when an hourly cron job does not exist" do
+      before do
+        @test_hourly_cron = "#{Dir.tmpdir}/miq-pg-maintenance-hourly.cron"
+        stub_const("ApplianceConsole::DatabaseMaintenance::HOURLY_CRON", @test_hourly_cron)
+      end
+
+      after do
+        FileUtils.rm_f(@test_hourly_cron)
+      end
+
+      it "adds a new hourly cron job" do
+        expected_cron_file = "#!/bin/sh\n"
```

Comment:
Can we make this a `let` with a heredoc? I think that would be a lot easier to read.